### PR TITLE
Add Progressive Enhancement

### PR DESCRIPTION
**Pull request by jbmoelker**
**https://github.com/jbmoelker/progressive-enhancement-resources**

**A comprehensive collection of resources on Progressive Enhancement. From concept and strategies to feature detection and testing methods. Complete with a list of (code) examples.**

~~A few notes:~~

- ~~Collection has a Table of Contents section, but it's named `Table of Contents`, not `Contents`.~~
- ~~Items are formatted consistently, but descriptions start with a lowercase.~~
- ~~Collection does not have the Awesome badge yet as I figured I'd only add this when the list is accepted.~~

~~Willing to change the above if needed.~~

# By submitting this pull request I confirm I've read and complied with the below requirements.

Failure to properly do so will just result in the pull request being closed and everyone's time wasted. Please read it twice. Most people miss many things.

- I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title. For example, `Add Name of List`, not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should include a short description about the list project/theme (not the list itself), and should be added at the bottom of the appropriate category.
- The list I added
	- **has been around for at least 30 days,**
	- is the result of hard work and is the best I could possibly produce,
	- is a non-generated Markdown file in a GitHub repo,
	- is not a duplicate,
	- includes a succinct description of the project/theme at the top ([example](https://github.com/willempienaar/awesome-quantified-self)),
	- includes a project logo/illustration (whenever possible) at the top right of the readme ([example](https://github.com/sindresorhus/awesome-electron)). (The logo should link to the project website or any relevant website. The logo should be high-DPI, so set it to maximum half the width of the original image.),
	- only has awesome items **with descriptions** *(Awesome lists are curations of the best, not of everything)*,
	- contains the [awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge) on the right side of the list heading,
	- has a Table of Contents section **named `Contents`** as the first section,
	- has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license) *(Meaning something like CC0, NOT a code licence like MIT, BSD, Apache, etc)* *(If you use a license badge, it should be SVG, not PNG)*,
	- has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines),
	- doesn't include a Travis badge,
	- has been checked for inconsistent formatting, spelling, and grammar:
		- Each link description starts with an uppercase character and ends with a period.
		- Drop all the `A` / `An` prefixes in the descriptions.
		- Consistent naming. For example, `Node.js`, not `NodeJS` or `node.js`.

### CODE REVIEW

**Comment by jbmoelker** [comment]
@sindresorhus and other reviewers, do you have any feedback? Would love to get this merged.

**Comment by legodude17** [comment]
Not followed:
 * Each link description starts with an uppercase character and ends with a period.
 * has a Table of Contents section named Contents as the first section,
 * has an appropriate license (Meaning something like CC0, NOT a code licence like MIT, BSD, Apache, etc) (If you use a license badge, it should be SVG, not PNG),
 * contains the awesome badge on the right side of the list heading,

**Comment by jbmoelker** [comment]
@legodude17 thanks for the response. I agree with most of your feedback as I pointed out in [my initial PR description](https://github.com/sindresorhus/awesome/pull/916#issue-216021408):

> A few notes:
> * Collection has a Table of Contents section, but it's named Table of Contents, not Contents.
> * Items are formatted consistently, but descriptions start with a lowercase.
> * Collection does not have the Awesome badge yet as I figured I'd only add this when the list is accepted.
>
> Willing to change the above if needed.

So should I take from your feedback, that yes, I must change these things even though they are currently consistent within the list itself?

Regarding the license, [it already has one](https://github.com/jbmoelker/progressive-enhancement-resources#license), very similar to [the one of this repository](https://github.com/sindresorhus/awesome#license). I don't quite understand what you would like me to change there? Should I also add it as a separate `license.md`?

**Comment by legodude17** [comment]
@jbmoelker 

> So should I take from your feedback, that yes, I must change these things even though they are currently consistent within the list itself?

Yes

> Regarding the license, it already has one, very similar to the one of this repository. I don't quite understand what you would like me to change there? Should I also add it as a separate license.md?

Yep

**Comment by jbmoelker** [comment]
@legodude17 clear. I updated to `Contents` and `[Title of link](url) - Short description.` and added a license and badge.

**Comment by legodude17** [comment]
@jbmoelker Yay! Suggestions: capitalize the "r" in "resources" in the heading.

**Comment by jbmoelker** [comment]
@legodude17 sure. Done.

**Comment by legodude17** [comment]
@jbmoelker Just noticed, all the headings should be in title case (Eg: "Feature Detection" instead of "Feature detection").

**Comment by jbmoelker** [comment]
@legodude17 I've changed all headings and table of contents entries to Title Case.

File: readme.md [hunk 1]
```diff
@@ -207,6 +207,7 @@
 - [Sass](https://github.com/Famolus/awesome-sass) - CSS preprocessor.
 - [Ant Design](https://github.com/websemantics/awesome-ant-design) - Enterprise-class UI design language.
 - [Less](https://github.com/LucasBassetti/awesome-less) - CSS preprocessor.
+- [Progressive Enhancement](https://github.com/jbmoelker/progressive-enhancement-resources) - From concept and strategies to feature detection & testing methods. Complete with a list of (code) examples.
```
Comment (sindresorhus) [inline readme.md]: > The entry in the Awesome list should include a short description about the list project/theme **(not the list itself)**

:arrow_up: and needs to be shorter.

**Comment by sindresorhus** [comment]
The list looks pretty good. Maybe link to some good YouTube videos?

The repo should have the following GitHub topics set: `awesome-list`, `awesome`, `list`.

The "progressive enhancement" part of `Progressive enhancement means gradually improving the user experience` should link to its Wikipedia page: https://en.wikipedia.org/wiki/Progressive_enhancement

**Comment by sindresorhus** [comment]
> Should I also add it as a separate license.md?
> > Yep

@legodude17 That is not a requirement.

File: readme.md (hunk 1 above)
Comment (jbmoelker) [inline readme.md]: Fair enough. `Progressive Enhancement - Concept, strategies, feature detection and (code) examples.` better? That's a description of 59 characters. Longest in the list is currently just over 100 characters. I don't know what's acceptable.

**Comment by jbmoelker** [comment]
@sindresorhus 

> The list looks pretty good. Maybe link to some good YouTube videos?

I'm open to it. I don't know any good ones of the top of my head.

> The repo should have the following GitHub topics set: awesome-list, awesome, list.

Done.

> The "progressive enhancement" part of Progressive enhancement means gradually improving the user experience should link to its Wikipedia page: https://en.wikipedia.org/wiki/Progressive_enhancement

Done.

File: readme.md (hunk 1 above)
Comment (sindresorhus) [inline readme.md]: It's not about the exact amount of characters, but rather being as succinct as possible. Please also read what a clearly bolded. The description should be about what progressive enhancement is, not what your list contains.

Comment (jbmoelker) [inline readme.md]: Sorry, that part didn't register with me until now. I find it very hard to describe the concept of PE in very few words. How about I just remove the description entirely?

**Comment by jbmoelker** [comment]
@legodude17 @sindresorhus I updated the list with all given feedback, removed the description and resolved the merge conflict. So I think it's ready to merge.

**Comment by legodude17** [comment]
GitHub says there is still a merge conflict.

**Comment by jbmoelker** [comment]
@legodude17 yes, that's because after I resolved it the `preact` PR was merged causing a new conflict. So resolved it again.

**Pull request merged**